sphinx/getting_started.md

```diff
@@ -18,67 +18,90 @@ pip install guppylang
 
 
 @guppy
-def teleport(src: qubit @owned, tgt: qubit) -> None:
-    """Teleports the state in `src` to `tgt`."""
-    # Create ancilla and entangle it with src and tgt
-    tmp = qubit()
-    h(tmp)
-    cx(tmp, tgt)
-    cx(src, tmp)
-
-    # Apply classical corrections
-    h(src)
-    if measure(src):
-        z(tgt)
-    if measure(tmp):
-        x(tgt)
+def simple_circuit() -> qubit:
+    q1, q2 = qubit(), qubit()
+
+    h(q1)
+    cx(q1, q2)
+
+    outcome = measure(q1)
+    result("q1", outcome)
+
+    if outcome:
+        x(q2)
+
+    return q2
+simple_circuit.check()
 ```
-As the input `src` qubit is measured and consumed, the teleport function must take [ownership](language_guide/ownership.md) of this qubit. We indicate that the function takes ownership of the `src` qubit using the `@owned` annotation.
 
-For execution, we can write a function to teleport the $|1\rangle$ state to serve as the entry point to our quantum program.
+For execution, we can write a function that invokes the circuit and consumes the produced qubit via a measurement.
+The outcome is recorded for later evaluation as well.
 
 ```{code-cell} ipython3
-from guppylang.std.builtins import result
-
 @guppy
-def teleport_one_state() -> None:
-    src = qubit()
-    tgt = qubit()
+def evaluate() -> None:
+    q = simple_circuit()
+    result("q2", measure(q))
+```
 
-    # Let's teleport the |1> state
-    x(src)
-    teleport(src, tgt)
+Finally, we can emulate our circuit implementation using the stabilizer simulator. 
+Our program is executed for a single shot using the `run` method.
 
-    result("teleported", measure(tgt))
+```{code-cell} ipython3
+emulator = evaluate.emulator(n_qubits=2).stabilizer_sim().with_seed(3)
+sim_result = emulator.run()
+list(sim_result.results)
 ```
 
+However, this is only a single shot.
+We can produce larger numbers of shots using the same API by changing the configuration of the emulator.
+Let's do this and plot the results using `matplotlib`:
+```{code-cell} ipython3
+import matplotlib.pyplot as plt
+import numpy as np
+
+shots = evaluate.emulator(n_qubits=2).with_seed(0).with_shots(10000).run()
 
-Finally, we can emulate our teleportation program using the stabilizer simulator. 
-Our program is executed for a single shot using the `run` method.
+fig, ax = plt.subplots(1, 1)
+possible_outcomes = ["00", "01", "10", "11"]
+idx = np.asarray(list(range(len(possible_outcomes))))
 
-```{code-cell} ipython3
+ax.bar(idx, [len([1 for shot in shots if str(shot.as_dict()['q1']) + str(shot.as_dict()['q2']) == o]) for o in possible_outcomes])
 
-sim_result = teleport_one_state.emulator(n_qubits=3).stabilizer_sim().with_seed(2).run()
-list(sim_result.results)
+ax.set_title("Circuit simulation")
+ax.set_xlabel("Measurement outcomes")
+ax.set_xticks(idx)
+ax.set_xticklabels(possible_outcomes)
+ax.set_ylabel("Frequency")
+plt.show()
 ```
+The results roughly match the expected distribution that was predicted in the circuit.
```

Comment:
I think it would also be nice to print the counts of the simulation so we could see how the counts correspond to the bar chart

Reply:
<img width="856" height="394" alt="image" src="https://github.com/user-attachments/assets/2c1a8893-8439-4ee6-b322-ba935840c490" />